# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Davidsoooooon/ML-Intern---Briones/blob/main/work/notebooks/w01_research_question.ipynb)

This notebook frames my Week 1 capstone direction before any modeling.

## 1. My lane (or freestyle) and why

I chose **Lane 2: Refresh / Content Opportunity Scoring** as my provisional lane. My goal is to help a content editor decide which pages should be reviewed first when time is limited. The unit of analysis is one anonymized content page. This lane fits the starter data because it contains measured signals such as impressions, clicks, average position, CTR, content age, freshness, engagement, and trend direction. These signals can support a ranked review queue rather than treating every page as equally urgent.

## 2. The question: decision, action, cost of a wrong call

**Research question:** Among pages with enough evidence, which pages should an editor review first for refresh, expansion, protection, or monitoring?

**Decision:** Which pages should enter the editor's top review queue?

**Output:** A ranked list of anonymized pages with a priority score, suggested action, confidence level, and understandable reason codes.

**Action:** A content editor reviews the highest-ranked pages and decides whether to refresh, expand, protect, monitor, or leave each page unchanged.

**Cost of a wrong call:** A false positive could waste editing time or lead to unnecessary changes on a healthy page. A false negative could cause the team to miss a page that is losing useful visibility or traffic.

**Why data or ML may help:** Several signals may interact in ways that are difficult to represent with one fixed rule. However, the goal is not simply to train a model. I will compare any learned method with a transparent rule-based baseline and keep the simpler approach unless the model improves the actual ranking decision.

## 3. Quick look at the data (2-3 real numbers)

The following cell loads the starter CSV and calculates a few numbers that support this lane choice.

In [1]:
from pathlib import Path
import pandas as pd

local_paths = [
    Path("data/raw/content_refresh_anonymized.csv"),
    Path("../../data/raw/content_refresh_anonymized.csv"),
]

csv_path = next((p for p in local_paths if p.exists()), None)
if csv_path:
    df = pd.read_csv(csv_path)
else:
    url = (
        "https://raw.githubusercontent.com/Davidsoooooon/"
        "ML-Intern---Briones/main/data/raw/"
        "content_refresh_anonymized.csv"
    )
    df = pd.read_csv(url)

declining = (df["trend_direction"] == "down").sum()

print(f"Starter rows: {len(df):,}")
print(f"Declining pages: {declining:,}")
print(f"Declining rate: {declining / len(df):.1%}")
print(f"Pseudonymized clients: {df['client_id'].nunique():,}")

Starter rows: 30,000
Declining pages: 16,262
Declining rate: 54.2%
Pseudonymized clients: 32


### Why these numbers support the lane

The starter dataset contains **30,000 anonymized pages across 32 pseudonymized clients**. In the current snapshot, **16,262 pages, or 54.2%, are marked as trending down**. This does not mean that every declining page should automatically be edited, but it shows that there are many possible review candidates. Because an editor cannot inspect all of them at once, a transparent ranking system could help focus limited review time on the pages with the strongest evidence and highest potential value.

## 4. Careful words: what I can and can't claim

This project can describe **observed associations**, **directional patterns**, and **decision-support rankings** in the anonymized dataset. I may report that certain measured signals are associated with a page receiving higher review priority, or that one ranking method performs better than a baseline under a stated validation design.

I cannot claim that I discovered Google's algorithm, that a recommended refresh will cause traffic to recover, or that every page marked as declining truly needs an edit. The current starter label is based on the same snapshot and should be treated as a provisional proxy rather than a final future outcome. A stronger later version should use earlier measurements as features and a separate future window as the target, with client-holdout or time-aware validation.

## Self-check

- [x] I selected one predefined lane and explained why it fits.
- [x] I named the decision, output, action, and cost of a wrong call.
- [x] I showed at least two real numbers calculated from the starter data.
- [x] I explained why this is a decision-support problem, not just "train a model."
- [x] I used careful language and did not make causal or Google-algorithm claims.
- [x] I kept client names, URLs, and private queries out of the notebook.